### PR TITLE
Sql id changes

### PR DESCRIPTION
**Pull request by venkatsridhar95**
Changes to retrieve sql_id from Oracle. This will be added as part of the EXEC CAL message. It would be easier to map SQLHash -> Oracle sql_id

Note: The flag (OCI_PREP2_GET_SQL_ID) used is newly introduced and is not backward compatible with the 11g client.

### CODE REVIEW

**Review by shtiencode — COMMENTED** [review]
Do we need to remove DO_OCI_HANDLE_FREE macro and real_oci_handle_free() given that OCIHandleFree() is avoided when using OCIStmtPrepare2() per [oracle doc](https://docs.oracle.com/en/database/oracle/oracle-database/19/lnoci/performance-topics.html#GUID-49C78457-8F76-48E1-A2CD-7EB22175042F), it looks dead code here.

**Comment by venkatsridhar95** [comment]
We need real_oci_handle_free() for other handles (service context handle, error handle, and a few others). Stmt handle alone will use OCIStmtRelease()

**Comment by shtiencode** [comment]
> We need real_oci_handle_free() for other handles (service context handle, error handle, and a few others). Stmt handle alone will use OCIStmtRelease()

Got it, thanks. Perhaps add a comment around the macro not to use for statement handle.

**Review by shtiencode — APPROVED** [review]
Change looks good.

**Comment by venkatsridhar95** [comment]
Pls do not merge. I'll move the existing main to the internal mirror before merging this PR.

File: worker/cppworker/worker/OCCChild.cpp
Comment (ModEtchFill) [inline worker/cppworker/worker/OCCChild.cpp:1046]: Perhaps do in PREPARE with m_bind_data.clear()

**Review by ModEtchFill — APPROVED** [review]

**Pull request closed without merging**
